StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift

```diff
@@ -8,8 +8,54 @@
 import Foundation
 @_spi(STP) import StripeCore
 
-protocol FinancialConnectionsAPIClient {
+class FinancialConnectionsAPIClient {
+    let backingAPIClient: STPAPIClient
 
+    var isLinkWithStripe: Bool = false
+    var consumerPublishableKey: String?
+    var consumerSession: ConsumerSessionData?
+
+    static let shared = FinancialConnectionsAPIClient(apiClient: .shared)
+
+    init(apiClient: STPAPIClient) {
+        self.backingAPIClient = apiClient
+    }
+
+    func consumerPublishableKeyProvider(canUseConsumerKey: Bool) -> String? {
+        guard canUseConsumerKey, isLinkWithStripe, consumerSession?.isVerified == true else {
+            return nil
+        }
+        return consumerPublishableKey
+    }
+
+    func get<T: Decodable>(
+        resource: String,
+        parameters: [String: Any],
+        canUseConsumerPublishableKey: Bool
+    ) -> Promise<T> {
+        let consumerPubKey = consumerPublishableKeyProvider(canUseConsumerKey: canUseConsumerPublishableKey)
+        return backingAPIClient.get(
+            resource: resource,
+            parameters: parameters,
+            ephemeralKeySecret: consumerPubKey
```

Comment:
feel free to ignore 100% of this:

if you do end up changing some naming, the word "provider" sounds sort of ambiguous 

if some renaming is done, maybe this could all be written as:

```
        return backingAPIClient.get(
            resource: resource,
            parameters: parameters,
            ephemeralKeySecret: retrieveConsumerPublishableKeyIfNeeded(useConsumerPublishableKeyIfNeeded)
         )
```

some interesting things to note is that when one reads `ephemeralKeySecret: consumerPubKey` or `let consumerPubKey =` it makes it seem that "consumer publishable key" IS there because the "context of the computation of whether the key will be returned is elsewhere"

Reply:
>some interesting things to note is that when one reads ephemeralKeySecret: consumerPubKey or let consumerPubKey = it makes it seem that "consumer publishable key" IS there because the "context of the computation of whether the key will be returned is elsewhere"

Makes sense! In these scenarios, I'm a fan of the `possible` nomenclature: 
```swift
let possibleConsumerPublishableKey = consumerPublishableKeyProvider(canUseConsumerKey: useConsumerPublishableKeyIfNeeded)
return backingAPIClient.get(
    resource: resource,
    parameters: parameters,
    ephemeralKeySecret: possibleConsumerPublishableKey
)
```

Let me know what you think!

Reply:
yea all of that sounds good! this part isn't too critical and is very debatable haha